test/unit/puppetlabs/services/versioned_code_service/versioned_code_core_test.clj

```diff
@@ -58,3 +58,24 @@
         (testing "witout stderr"
           (let [result (vc-core/execute-code-content-script! (script-path "echo") environment code-id file-path)]
             (is (= (format "%s %s %s\n" environment code-id file-path) (IOUtils/toString result "UTF-8")))))))))
+
+(deftest valid-code-id?-test
+  (testing "valid-code-id? accepts valid code-ids"
+    (is (vc-core/valid-code-id? nil))
+    (is (vc-core/valid-code-id? "dcf16ec"))
+    (is (vc-core/valid-code-id? "master-plan:stage_1:destroy-all-humans")))
+
+  (testing "valid-code-id? rejects invalid code-ids"
+    (is (not (vc-core/valid-code-id? "bad code id")))
+    (is (not (vc-core/valid-code-id? "don't do it!")))
```

Comment:
Might want to remove the spaces here since those are tested in the last assertion, and this method should return `false` if even one invalid character shows up. Might also be worth testing `'` and `!` in separate tests